extensions/3rdparty/ArrayExtension/ArrayExtension.php

```diff
@@ -287,7 +287,7 @@ function arraysize( &$parser, $arrayid) {
 *   See: http://www.php.net/manual/en/function.array-search.php
 *   note it is extended to support regular expression match and index
 */
-    function arraysearch( &$parser, $arrayid, $needle, $index=0, $yes=null, $no=null) {
+    function arraysearch( $parser, $arrayid, $needle, $index=0, $yes=null, $no=null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$parser". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)